titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/common/rest/TitusExceptionMapper.java

```diff
@@ -226,9 +219,26 @@ private Response fromJobManagerException(JobManagerException e) {
             case TaskNotFound:
                 errorBuilder.status(HttpServletResponse.SC_NOT_FOUND);
                 break;
-            case NotServiceJob:
+            // these below are mapped to Status.FAILED_PRECONDITION for gRPC calls
+            case JobTerminating:
```

Comment:
We should not use internal server error for these. You can use status code 412 precondition failed.

Reply:
Will do. As discussed in person, the status code 412 is not really intended to be used for things like this (it is more related to headers passed in the request), but I think it is OK to put these errors into the 4xx category.